splitgpg2/__init__.py

```diff
@@ -458,7 +458,22 @@ async def command_AGENT_ID(self, untrusted_args: Optional[bytes]):
 
     async def command_HAVEKEY(self, untrusted_args: Optional[bytes]):
         # upper keygrip limit is arbitary
-        args = self.verify_keygrip_arguments(1, 200, untrusted_args)
+        if untrusted_args.startswith(b'--list'):
+            if b'=' in untrusted_args:
+                try:
+                    limit = int(untrusted_args[len(b'--list='):])
```

Comment:
Should a regex check and/or a length limit be used here?

Reply:
I don't think that's necessary, but I do need to check lower bound.

Reply:
I mostly want to guard against some huge number being produced.  A limit of 20 should be sufficient.  I would also check that the number is decimal and not negative.

Reply:
>  the number is decimal

I don't think there is a need for that. The actual value sent to gpg-agent is `limit` set here, regardless of its original format.